test/collect_coverage_api_test.dart

```diff
@@ -92,6 +101,11 @@ Future<Map<String, dynamic>> _collectCoverage(
     }
   });
   final Uri serviceUri = await serviceUriCompleter.future;
+  Set<String> isolateIdSet;
+  if (isolateIds) {
+    isolateIdSet = Set<String>.from(<String>[]);
+  }
 
-  return collect(serviceUri, true, true, false, scopedOutput, timeout: timeout);
+  return collect(serviceUri, true, true, false, scopedOutput,
+      timeout: timeout, isolateIds: isolateIdSet);
```

Comment:
This is a great start, but we should also add a test wherein we run with multiple isolates and only collect for one of them so that we don't accidentally break/regress this in future.

Reply:
Ok, I can look into doing that sometime today! I'll ping you when I have something figured out.

Reply:
@cbracken it looks like the test app was already spawning an isolate. I added a test to test against just that isolate instead of testing against both that isolate and root, and everything looks good. Let me know if you want any further changes!